library/datasort.c

```diff
@@ -1050,6 +1055,16 @@ static struct datasort_chunk *datasort_merge(struct datasort_ctl *ds_ctl)
 
 	EBLOB_WARNX(dcfg->log, EBLOB_LOG_INFO, "defrag: merge: start");
 
+	/* Calculate size of data and number of records */
+	for (index = 0; index != dcfg->bctl_cnt; ++index) {
+		struct eblob_base_ctl *bctl = dcfg->bctl[index];
+		err = eblob_base_setup_data(bctl);
```

Comment:
let's setup data when it is first required

Reply:
changed on blob's stats